H/HDF4/build_tarballs.jl

```diff
@@ -3,47 +3,51 @@
 using BinaryBuilder, Pkg
 
 name = "HDF4"
-version = v"4.2.16"
+version = v"4.3.0"
 
 # Collection of sources required to complete build
 sources = [
-    ArchiveSource("https://support.hdfgroup.org/ftp/HDF/releases/HDF4.2.16-2/src/hdf-4.2.16-2.tar.gz", "a24b18312d421686031c2d66635f7d5abb2fe879f8a182b7e02797b0da8d1f6c")
+    ArchiveSource("https://github.com/HDFGroup/hdf4/releases/download/hdf$(version)/hdf$(version).tar.gz",
+                  "282b244a819790590950f772095abcaeef405b0f17d2ee1eb5039da698cf938b"),
 ]
 
 # Bash recipe for building across all platforms
 script = raw"""
 cd $WORKSPACE/srcdir/hdf*
-mkdir build
-cd build
-cmake -DCMAKE_INSTALL_PREFIX=${prefix} \
+cmake -B build -G Ninja \
+    -DCMAKE_INSTALL_PREFIX=${prefix} \
     -DCMAKE_TOOLCHAIN_FILE=${CMAKE_TARGET_TOOLCHAIN} \
     -DCMAKE_BUILD_TYPE=Release \
     -DTEST_LFS_WORKS_RUN=0 \
     -DH4_PRINTF_LL_TEST_RUN=0 \
-    -DH4_PRINTF_LL_TEST_RUN__TRYRUN_OUTPUT= \
-    ..
-make -j${nproc}
-make install
+    -DH4_PRINTF_LL_TEST_RUN__TRYRUN_OUTPUT=
+cmake --build build --parallel ${nproc}
+cmake --install build
 """
 
 # These are the platforms we will build for by default, unless further
 # platforms are passed in on the command line
 platforms = supported_platforms()
-# /opt/x86_64-unknown-freebsd13.2/x86_64-unknown-freebsd13.2/sys-root/usr/include/rpc/xdr.h:125:2: error: unknown type name 'u_int'
-filter!(!Sys.isfreebsd, platforms)
-# /workspace/srcdir/hdf-4.2.16-2/mfhdf/xdr/types.h:68:18: error: conflicting types for ‘u_quad_t’
-filter!(p -> !(libc(p) == "musl" && nbits(p) == 64), platforms)
 
 # The products that we will ensure are always built
 products = [
+    # ExecutableProduct("h4cc", :h4cc),   # `h4cc` is not built on Windows
+    ExecutableProduct("hdfed", :hdfed),
+    ExecutableProduct("hdfimport", :hdfimport),
+    ExecutableProduct("hdfls", :hdfls),
+    ExecutableProduct("hdiff", :hdiff),
+    ExecutableProduct("hdp", :hdp),
+    ExecutableProduct("hrepack", :hrepack),
+    ExecutableProduct("ncdump", :ncdump),
+    ExecutableProduct("ncgen", :ncgen),
     LibraryProduct("libhdf", :libhdf),
-    LibraryProduct("libmfhdf", :libmfhdf)
+    LibraryProduct("libmfhdf", :libmfhdf),
 ]
 
 # Dependencies that must be installed before this package can be built
 dependencies = [
-    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a"))
-    Dependency(PackageSpec(name="JpegTurbo_jll", uuid="aacddb02-875f-59d6-b918-886e6ef4fbf8"))
+    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a"); compat="1.3.1"),
```

Comment:
Zlib is complicated because it's bound to the julia version, if you require zlib 1.3.1 you can't install the package in all older versions of Julia.  For v1.6 (our baseline): https://github.com/JuliaLang/julia/blob/f9720dc2ebd6cd9e3086365f281e62506444ef37/stdlib/Zlib_jll/Project.toml#L3
```suggestion
    Dependency(PackageSpec(name="Zlib_jll", uuid="83775a58-1f1d-513f-b197-d71354ab007a"); compat="1.2.12"),
```